src/OpenLoco/src/Objects/ObjectManager.cpp

```diff
@@ -984,10 +984,30 @@ namespace OpenLoco::ObjectManager
         call(0x004C3A9E);
     }
 
+    // TODO: Refactor this, variable is also defined in PaintSurface.cpp.
+    static loco_global<LandObjectFlags[getMaxObjects(ObjectType::land)], 0x00F003D3> _landObjectFlags;
+
+    // 0x004697A1
+    static void updateLandObjectFlags()
+    {
+        for (size_t i = 0; i < getMaxObjects(ObjectType::land); i++)
+        {
+            auto* landObj = get<LandObject>(i);
+            if (landObj != nullptr)
+            {
+                _landObjectFlags[i] = landObj->flags;
+            }
+            else
+            {
+                _landObjectFlags[i] = LandObjectFlags::none;
+            }
```

Comment:
Not seeing this branch in the disasm, correct? It seems like a good idea, though.

Reply:
Yeah that is not in the original, it shouldn't be used when no surface/land references the object but given it loops over all objects might as well clear the state.